### PR TITLE
Unfeaturized entities

### PR DESCRIPTION
**Pull request by mauricedoepke**
Ported from rasa_core repository

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by mauricedoepke** [comment]
This is the port of: https://github.com/RasaHQ/rasa_core/pull/1841
@wochinge

File: data/test_domains/default_unfeaturized_entities.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - hey there!
+  utter_goodbye:
+    - goodbye :(
+  utter_default:
+    - default message
+
+actions:
+  - utter_default
+  - utter_greet
+  - utter_goodbye
```
Comment (wochinge) [inline data/test_domains/default_unfeaturized_entities.yml]: please add a newline

File: docs/core/domains.rst [hunk 2]
```diff
@@ -272,17 +272,26 @@ multiple responses and Rasa will randomly pick one of them, e.g.:
...
+        use_entities: None
 
-This means that entities for those intents will be unfeaturized and therefore
+To ignore some entities or explicitly take only certain entities
+into account you can use this syntax:
+
+.. code-block:: yaml
+
+  intents:
+    - greet: {include_entities: [name]}
```
Comment (wochinge) [inline docs/core/domains.rst]: Should we format that a bit nice and do?
```
intents:
- greet:
   include_entities: 
   - entity1
   - entity2

File: rasa/core/domain.py [hunk 3]
```diff
@@ -112,9 +112,9 @@ def from_dict(cls, data: Dict) -> "Domain":
         utter_templates = cls.collect_templates(data.get("templates", {}))
         slots = cls.collect_slots(data.get("slots", {}))
         additional_arguments = data.get("config", {})
-        intent_properties = cls.collect_intent_properties(data.get("intents", {}))
+        intent_list = data.get("intents", {})
```
Comment (wochinge) [inline rasa/core/domain.py]: Avoid hungarian notation
```suggestion
        intents = data.get("intents", {})
```

File: rasa/core/domain.py [hunk 4]
```diff
@@ -112,9 +112,9 @@ def from_dict(cls, data: Dict) -> "Domain":
...
         return cls(
-            intent_properties,
+            intent_list,
```
Comment (wochinge) [inline rasa/core/domain.py]:
```suggestion
            intents,
```

File: rasa/core/domain.py [hunk 5]
```diff
@@ -245,11 +245,16 @@ def collect_intent_properties(intent_list):
             if isinstance(intent, dict):
```
[the remark this answers is not in the document]
Comment (mauricedoepke) [reply rasa/core/domain.py]: Are your sure it is `Union[Text, Dict[Text, Any]]` ?
I'm not familiar with pythons type system yet, but I think:
`Union[List[Text], List[Dict[Text, Any]]]` would be the correct type if I interpret the `collect_intent_properties` correct.

File: rasa/core/domain.py [hunk 6]
```diff
@@ -481,11 +486,34 @@ def get_parsing_states(self, tracker: "DialogueStateTracker") -> Dict[Text, floa
         # be ignored for the current intent
         for entity in tracker.latest_message.entities:
```
[the remark this answers is not in the document]
Comment (mauricedoepke) [reply rasa/core/domain.py]: Thanks, good catch.

File: rasa/core/training/interactive.py [hunk 7]
```diff
@@ -839,16 +839,13 @@ def _intents_from_messages(messages):
     templates = NEW_TEMPLATES
 
     # TODO for now there is no way to distinguish between action and form
-    intent_properties = Domain.collect_intent_properties(
-        _intents_from_messages(messages)
-    )
```
Comment (wochinge) [inline rasa/core/training/interactive.py]: Todo belongs to  `collected_actions`
```suggestion
```

File: rasa/core/domain.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (wochinge) [reply rasa/core/domain.py]: Good catch! Totally missed the loop 🙈 
`List[Union[Text, Dict[Text, Any]]]`

**Comment by wochinge** [comment]
@mauricedoepke Let me know if you need any help with the remaining comments.

**Comment by wochinge** [comment]
@mauricedoepke I quickly fixed the type annotations and some unused imports. Let's see whether travis passes and then merge this 🚀

File: rasa/core/domain.py [hunk 8]
```diff
@@ -477,20 +478,47 @@ def get_parsing_states(self, tracker: "DialogueStateTracker") -> Dict[Text, floa
...
+        intent_name = latest_message.intent.get("name")
+        intent_config = self.intent_config(intent_name)
+        entities = latest_message.entities
+        entity_names = {
+            entity["entity"] for entity in entities if "entity" in entity.keys()
+        }
+
+        # `use_entities` is either a list of explicitly included entities
+        # or `True` if all should be included
+        include = intent_config.get("use_entities")
```
Comment (wochinge) [inline rasa/core/domain.py]: I think this should be `True` by default @mauricedoepke 
```suggestion
        include = intent_config.get("use_entities", True)
```

**Comment by wochinge** [comment]
@mauricedoepke Thanks a lot for all your work! Really cool to have this in Rasa 🚀 🥇 I quickly fixed the types and merged it before it outdates again.

**Pull request merged**